src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -131,6 +146,21 @@ public uint GetStoragePrice(IReadOnlyStore snapshot)
         }
 
         /// <summary>
+        /// Gets the block generation time in milliseconds.
+        /// This can only be called after the HF_Echidna.
+        /// </summary>
+        /// <param name="snapshot">The snapshot used to read data.</param>
+        /// <returns>The block generation time in milliseconds.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public uint GetBlockGenTime(IReadOnlyStore snapshot)
+        {
+            var item = snapshot.TryGet(_blockGenTime, out var value) ? value : null;
+            if (item is null) return DefaultBlockGenTime;
```

Comment:
It can't be null, we initialize this value at Echidna's start height. Just use
```
 return (uint)(BigInteger)snapshot[_blockGenTime];
```

Reply:
> It can't be null, we initialize this value at Echidna's start height. Just use
> 
> ```
>  return (uint)(BigInteger)snapshot[_blockGenTime];
> ```

It won't be null. It's declared at the begining.

Reply:
can be a future optimization?

Reply:
No, just update the code.

Reply:
any particular reason of must update now?

Reply:
We need to follow the behaviour of other system settings setters (like `GetStoragePrice`, `GetExecFeeFactor` and `GetFeePerByte`.  The value of `blockGenTime` must be stored in the Policy storage during Echidna-level contract initialization. If (for some reason) this value is missing from the storage then it's a bug in the Policy contract and we need to fix this bug instead of returning default value.